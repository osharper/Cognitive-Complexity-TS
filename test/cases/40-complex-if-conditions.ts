function simpleSingleCondition() {
    if (true) {
        // do something
    }
}

function singleBooleanNegation() {
    if (!true) {
        // do something
    }
}

function multipleAndConditions() {
    if (condition1 && condition2 && condition3) {
        // do something
    }
}

function multipleOrConditions() {
    if (condition1 || condition2 || condition3) {
        // do something
    }
}

function mixedAndOrConditions() {
    if (condition1 && condition2 || condition3 && condition4) {
        // do something
    }
}

function complexParenthesizedConditions() {
    if (condition1 && (condition2 || condition3) && condition4) {
        // do something
    }
}

function negatedComplexConditions() {
    if (val > someConstant && (condition2 || condition3) && !condition4 && !condition5 || val < someOtherConstant) {
        // do something
    }
}

function elseIfWithComplexConditions() {
    if (val > someConstant && condition2) {
        // do something
    } else if (val < someOtherConstant || !condition3) {
        // do something else
    }
}

let condition1, condition2, condition3, condition4, condition5, val, someConstant, someOtherConstant;