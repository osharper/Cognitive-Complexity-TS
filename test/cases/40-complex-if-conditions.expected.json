{
    "40-complex-if-conditions.ts": {
        "score": 29,
        "inner": [
            {
                "name": "simpleSingleCondition",
                "score": 1,
                "line": 1,
                "column": 1
            },
            {
                "name": "singleBooleanNegation",
                "score": 2,
                "line": 7,
                "column": 1
            },
            {
                "name": "multipleAndConditions",
                "score": 2,
                "line": 13,
                "column": 1
            },
            {
                "name": "multipleOrConditions",
                "score": 2,
                "line": 19,
                "column": 1
            },
            {
                "name": "mixedAndOrConditions",
                "score": 5,
                "line": 25,
                "column": 1
            },
            {
                "name": "complexParenthesizedConditions",
                "score": 4,
                "line": 31,
                "column": 1
            },
            {
                "name": "negatedComplexConditions",
                "score": 8,
                "line": 37,
                "column": 1
            },
            {
                "name": "elseIfWithComplexConditions",
                "score": 5,
                "line": 43,
                "column": 1
            }
        ]
    }
}